### PR TITLE
Bump Prettier from 2.8.8 to 3.0.0

### PR DESCRIPTION
**Pull request by queengooborg**
This PR bumps Prettier from 2.8.8 to 3.0.0, and then fixes issues with the content commands caused by changes to the Prettier API.  In Prettier v3.0, the `format()` command became asynchronous, so we were returning a Promise every so often, which caused errors when running commands.  This PR aims to fix this by adding an `await`, and making the rest of the function chain asynchronous.

**Note:** because the content repositories use Prettier 3.0.0, these commands are **broken** regardless of the Prettier version Yari is requesting.

**Note:** because Prettier has changed its defaults, the formatting output has also changed.  TO pass linting, all files had to be formatted, which unfortunately means a greater risk of merge conflicts.  Since this is a critical bug fix, it should be merged quickly anyways.

### CODE REVIEW

**Comment by queengooborg** [comment]
I'm marking this as a draft because the CI test is failing and I don't have the availability to dig into it further.  CC @caugner @fiji-flo @LeoMcA (note: this is somewhat urgent as it is disrupting contributor workflow)

**Comment by queengooborg** [comment]
Oh...well, apparently, the last commit fixed it all up.  😛

**Comment by queengooborg** [comment]
@LeoMcA, I see that you're around -- are you available and willing to review and merge this quickly by chance?  (Highly recommend reviewing by commit since Prettier formatting also has to be run in the same go...)

**Comment by LeoMcA** [comment]
@queengooborg taking a look, but first up: I reckon now's not the time to make the [`trailingComma` change](https://prettier.io/blog/2023/07/05/3.0.0.html#change-the-default-value-for-trailingcomma-to-all-11479httpsgithubcomprettierprettierpull11479-by-fiskerhttpsgithubcomfisker-13143httpsgithubcomprettierprettierpull13143-by-sosukesuzukihttpsgithubcomsosukesuzuki) if we want to merge this quickly (and if we do that in a separate PR, [we can add that squashed commit to a blame ignore revs file](https://docs.github.com/en/repositories/working-with-files/using-files/viewing-a-file#ignore-commits-in-the-blame-view)).

Could you configure `{ "trailingComma": "es5" }` and force push with a new "run prettier" commit (or none if there's no other changes between 2.8.8 and 3)?

**Comment by queengooborg** [comment]
That's a fair point, since most of the formatting changes are from adding trailing commas.  There are still some other changes, so I've included that commit.

File: content/document.ts [hunk 1]
```diff
@@ -46,7 +46,7 @@ const getHTMLPath = (folder: string) => path.join(folder, HTML_FILENAME);
 const getMarkdownPath = (folder: string) =>
   path.join(folder, MARKDOWN_FILENAME);
 
-export function updateWikiHistory(
+export async function updateWikiHistory(
```
Comment (LeoMcA) [inline content/document.ts:49]: Also needs to be `await`ed here:

https://github.com/mdn/yari/blob/5a9af2b3781562425ca57149a76cb0eb8563d457/tool/sync-translated-content.ts#L209

File: content/document.ts [hunk 2]
```diff
@@ -355,7 +355,7 @@ export const read = memoize((folderOrFilePath: string, ...roots: string[]) => {
   };
 });
 
-export function update(url: string, rawBody: string, metadata) {
+export async function update(url: string, rawBody: string, metadata) {
```
Comment (LeoMcA) [inline content/document.ts:358]: Also needs to be `await`ed here:
https://github.com/mdn/yari/blob/5a9af2b3781562425ca57149a76cb0eb8563d457/content/document.ts#L576

File: build/blog.ts
Comment (LeoMcA) [inline build/blog.ts:1]: Bit of weirdness going on in this file, we've formatted from `main` without the merge commit

**Review by LeoMcA — CHANGES_REQUESTED** [review]

File: client/src/playground/index.tsx [hunk 3]
```diff
@@ -1,10 +1,11 @@
...
 import useSWRImmutable from "swr/immutable";
-import prettier from "prettier/esm/standalone.mjs";
-import parserBabel from "prettier/esm/parser-babel.mjs";
-import parserCSS from "prettier/esm/parser-postcss.mjs";
-import parserHTML from "prettier/esm/parser-html.mjs";
+import prettier from "prettier/standalone.mjs";
+import prettierPluginBabel from "prettier/plugins/babel.mjs";
+import prettierPluginCSS from "prettier/plugins/postcss.mjs";
+import prettierPluginESTree from "prettier/plugins/estree.mjs";
+import prettierPluginHTML from "prettier/plugins/html.mjs";
```
[the remark this answers is not in the document]
Comment (queengooborg) [reply client/src/playground/index.tsx:8]: Looks like the `estree` plugin is a little broken right now: https://github.com/prettier/prettier/pull/15018

All good to keep that one as `.mjs` for now?

File: build/blog.ts
Comment (queengooborg) [reply build/blog.ts:1]: Slight issue with force pushing; should be all fixed now!

File: client/src/playground/index.tsx [hunk 4]
```diff
@@ -1,10 +1,12 @@
...
-import prettier from "prettier/esm/standalone.mjs";
-import parserBabel from "prettier/esm/parser-babel.mjs";
-import parserCSS from "prettier/esm/parser-postcss.mjs";
-import parserHTML from "prettier/esm/parser-html.mjs";
+import prettier from "prettier/standalone.js";
+import prettierPluginBabel from "prettier/plugins/babel.js";
+import prettierPluginCSS from "prettier/plugins/postcss.js";
+// XXX Using .mjs until https://github.com/prettier/prettier/pull/15018 is deployed
+import prettierPluginESTree from "prettier/plugins/estree.mjs";
+import prettierPluginHTML from "prettier/plugins/html.js";
```
Comment (LeoMcA) [inline client/src/playground/index.tsx:9]: Sorry, I meant drop the extension altogether, like:

```suggestion
import prettier from "prettier/standalone";
import prettierPluginBabel from "prettier/plugins/babel";
import prettierPluginCSS from "prettier/plugins/postcss";
// XXX Using .mjs until https://github.com/prettier/prettier/pull/15018 is deployed
import prettierPluginESTree from "prettier/plugins/estree.mjs";
import prettierPluginHTML from "prettier/plugins/html";
```

**Review by LeoMcA — APPROVED** [review]
Looks good, just the one change:

**Pull request merged**